src/Microsoft.CorrelationVector/CorrelationVector.cs

```diff
@@ -1,397 +1,159 @@
-            {
-                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, "Unsupported correlation vector version: {0}", version));
-            }
-        }
-
-        private static CorrelationVectorVersion InferVersion(string correlationVector, bool reportErrors)
-        {
-            int index = correlationVector == null ? -1 : correlationVector.IndexOf('.');
-
-            if (CorrelationVector.BaseLength == index)
-            {
-                return CorrelationVectorVersion.V1;
-            }
-            else if (CorrelationVector.BaseLengthV2 == index)
-            {
-                return CorrelationVectorVersion.V2;
-            }
-            else
-            {
-                //By default not reporting error, just return V1
-                return CorrelationVectorVersion.V1;
-            }
-        }
-
-        private static bool IsImmutable(string correlationVector)
+                //By default not reporting error, just return V1
+                return CorrelationVectorVersion.V1;
+            }
+        }
+
+        /// <summary>
+        /// Converts a string representation of a Correlation Vector into this class.
+        /// </summary>
+        /// <param name="correlationVector">String representation</param>
+        /// <returns>The Correlation Vector based on its version.</returns>
+        public static CorrelationVector Parse(string correlationVector)
+        {
+            CorrelationVectorVersion version = InferVersion(correlationVector);
+            return RunStaticMethod(correlationVector, version, CorrelationVectorV1.Parse, CorrelationVectorV2.Parse);
+
+        }
+
+        public static CorrelationVector Extend(string correlationVector)
+        {
+            CorrelationVectorVersion version = InferVersion(correlationVector);
+            return RunStaticMethod(correlationVector, version, CorrelationVectorV1.Extend, CorrelationVectorV2.Extend);
+
+        }
+
+        public static CorrelationVector Spin(string correlationVector)
+        {
+            SpinParameters defaultParameters = new SpinParameters
+            {
+                Interval = SpinCounterInterval.Coarse,
+                Periodicity = SpinCounterPeriodicity.Short,
+                Entropy = SpinEntropy.Two
+            };
+            return Spin(correlationVector, defaultParameters);
+        }
+
+        public static CorrelationVector Spin(string correlationVector, SpinParameters parameters)
         {
-            return !string.IsNullOrEmpty(correlationVector) && correlationVector.EndsWith(CorrelationVector.TerminationSign);
-        }
-
-        private static bool IsOversized(string baseVector, int extension, CorrelationVectorVersion version)
+            CorrelationVectorVersion version = InferVersion(correlationVector);
+            switch (version)
+            {
+                case CorrelationVectorVersion.V1:
+                    return NotHandledMethod(correlationVector);

```

Comment:
Let's keep things consistent and call the CorrelationVectorV1.Spin() method and let that implementation throw an exception. The logic for whether or not a certain method is supported in a certain version should be in the version implementation, not in the base class